### PR TITLE
CDAP-1895 fixing a bug where previewing a query would close the

### PR DESCRIPTION
**Pull request by albertshau**
query if there were no results returned.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT1265

**Comment by jguery** [comment]
Where do you prevent the query from not entering the inactive state? I don't see those changes.

**Comment by albertshau** [comment]
calls fetchNextResults instead of nextResults.  nextResults is the one that calls timeoutAggressively, which makes it inactive.

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java [hunk 1]
```diff
@@ -766,13 +764,8 @@ public QueryStatus getStatus(QueryHandle handle) throws ExploreException, Handle
...
-    Lock nextLock = getOperationInfo(handle).getNextLock();
-    nextLock.lock();
     try {
-      // Fetch results from Hive
-      LOG.trace("Getting results for handle {}", handle);
-      OperationHandle opHandle = getOperationHandle(handle);
-      List<QueryResult> results = fetchNextResults(opHandle, size);
+      List<QueryResult> results = fetchNextResults(handle, size);
       QueryStatus status = getStatus(handle);
       if (results.isEmpty() && status.getStatus() == QueryStatus.OpStatus.FINISHED) {
```
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: this is where the query handle is moved to inactive.

**Comment by albertshau** [comment]
tested on cdh4, cdh5, hdp2

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java (hunk 1 above)
Comment (jguery) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: Oh ok yes I see now. Thanks.

**Comment by jguery** [comment]
LGTM

**Pull request merged**